servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/CassandraBackendBuilder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.providers;
+
+import static org.projectnessie.quarkus.config.VersionStoreConfig.VersionStoreType.CASSANDRA;
+
+import javax.enterprise.context.Dependent;
+import org.projectnessie.versioned.storage.common.persist.Backend;
+
+@StoreType(CASSANDRA)
+@Dependent
+public class CassandraBackendBuilder implements BackendBuilder {
+
+  // @Inject CompletionStage<QuarkusCqlSession> client;
```

Comment:
If this is your only injection point of the driver session, maybe you don't need the Cassandra extension. Just create the session manually, and read the required properties (contact points and local DC) from some config properties like you are doing below to retrieve the keyspace.